### ISSUE TITLE
Uncaught undefined-variable error in try block

### ISSUE THREAD

**Issue by martimlobao** [PyCQA/pylint#3399]
### Steps to reproduce
1.
```python
# error.py
try:
    boop
except:
    pass
```
2. `pylint error.py`

### Current behavior
```
$ pylint error.py 
************* Module error
error.py:1:0: C0114: Missing module docstring (missing-module-docstring)
error.py:3:0: W0702: No exception type(s) specified (bare-except)

-----------------------------------
Your code has been rated at 5.00/10
```

### Expected behavior
`pylint` should raise an undefined error for `boop`
```
$ pylint error.py 
************* Module error
error.py:1:0: C0114: Missing module docstring (missing-module-docstring)
error.py:2:0: E0602: Undefined variable 'boop' (undefined-variable)
error.py:3:0: W0702: No exception type(s) specified (bare-except)

-----------------------------------
Your code has been rated at 5.00/10
```

### pylint --version output

```
pylint 2.4.4
astroid 2.3.3
Python 3.6.9 (default, Jan 24 2020, 16:11:17) 
[GCC 4.2.1 Compatible Apple LLVM 11.0.0 (clang-1100.0.33.12)]
```

**Comment by martimlobao** [comment]
As a workaround, I discovered that using `except Exception` instead of a bare `except` fixes the issue:
```python
try:
    boop
except:
    pass
```

**Comment by PCManticore** [comment]
This is properly working with the latest version from the `master` branch, which is going to be 2.5 when released:
```
pylint 2.5.0-dev1
astroid 2.4.0
Python 3.6.5 (default, May  7 2018, 10:07:07)
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

Thanks for the report!
